SignalServiceKit/src/Util/ParamParser.swift

```diff
@@ -121,6 +125,10 @@ public class ParamParser {
             throw ParseError.invalidFormat(key)
         }
 
+        guard data.count > 0 else {
+            return nil
```

Comment:
Wait, I'm not sure this is a good idea.  IMO there's a difference between a "missing/null" value and empty byte string.  Even if the field is optional.

Reply:
It's true that they are not necessarily the same thing.

I'm failing to think of a scenario where we "need" a data parameter, but accept a data parameter of 0 length.

Reply:
e.g. in this case the service is sending an empty string for receipts, which have no body.